### DISCUSSION TITLE
Web search results are not used by the LLM in response

### DISCUSSION THREAD

**Discussion by Hugobox** [open-webui/open-webui#3363] [General]
# Bug Report

## Description

**Bug Summary:**

After activating web search with Duckduckgo engine, LLM models do not seem to use the searched websites information when replying to the prompt.

**Steps to Reproduce:**
Activate web search, ask a question that has a recent updated answer, note that the reply is outdated, even if the web search results clearly indicate the right answer.

**Expected Behavior:**

The reply should take into account the updated information from the web search

**Actual Behavior:**

The reply contained outdated information.

## Environment

- **Open WebUI Version:** v0.3.4

## Reproduction Details

**Confirmation:**

- [x ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**

![image](https://github.com/open-webui/open-webui/assets/15982264/db610394-43ad-41a7-a6b0-f11c6899ec12)

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by Hugobox** [comment]
Is Web search working for other people? Does it take the webpage content and tries to RAG it via the embedding model?

**Reply by guenhter** [reply]
Yes. I use it all the time and it works like a charm.

![image](https://github.com/user-attachments/assets/1282c4e6-e981-489a-b27d-acb26304a43e)

**Comment by Hugobox** [comment]
@tjbck @justinh-rahb  Any help would be appreciated!

**Comment by fighter3005** [comment]
I actually noticed the same. I am new to the whole AI game, but I would have expected the web information to be included in the chat completion request (in the context xml brackets as described in RAG). However, the information seems to be missing, regardless wether the search query generation has been skipped (The skipping is a bit annoying).
![grafik](https://github.com/user-attachments/assets/7808f0fa-b1c5-4b64-a4f9-fd055479737c)

(The hole process of enabeling search specifically, and the search being perfomed for all prompts from there onwards seems also very unnatural. I guess we need to do a custom system prompt and let the model use functions to retrieve information from the web!?)

**Reply by tjbck** [reply]
Patched with [0.3.9 hotfix](https://github.com/open-webui/open-webui/pull/3943)

**Reply by changchiyou** [reply]
Facing the same problem in 0.3.10 using SearXNG:

![image](https://github.com/user-attachments/assets/992ee6df-d24b-438e-b032-4491b6c4cc95)

---

Based on the browse results attached below, my preliminary judgment (without thoroughly reviewing LiteLLM's logs and examining the relevant source code of Open WebUI) is that there is a flaw in the web crawling functionality within the built-in websearch feature of Open WebUI when handling search engine results. 

For instance, with the query "What is the latest released version of openwebui on github?", Open WebUI did find "https://github.com/open-webui/open-webui/releases," but the content it retrieved might have been read before the page was fully loaded, resulting in the omission of many important details that should have been included.

<details>
  <summary>Browse result of https://github.com/open-webui/open-webui/releases</summary>

  ```
  來源
  https://github.com/open-webui/open-webui/releases
  內容
  Releases ¬∑ open-webui/open-webui ¬∑ GitHub
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  Skip to content
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  Navigation Menu
  
  Toggle navigation
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  Sign in
  
  
  
  
  
  
  
  
  
  Product
  
  
  
  
  
  
  
  
  
  
  
  
  
  Actions
  Automate any workflow
  
  
  
  
  
  
  
  
  Packages
  Host and manage packages
  
  
  
  
  
  
  
  
  Security
  Find and fix vulnerabilities
  
  
  
  
  
  
  
  
  Codespaces
  Instant dev environments
  
  
  
  
  
  
  
  
  GitHub Copilot
  Write better code with AI
  
  
  
  
  
  
  
  
  Code review
  Manage code changes
  
  
  
  
  
  
  
  
  Issues
  Plan and track work
  
  
  
  
  
  
  
  
  Discussions
  Collaborate outside of code
  
  
  
  
  
  Explore
  
  
  
  All features
  
  
  
  
  
  Documentation
  
  
  
  
  
  
  
  GitHub Skills
  
  
  
  
  
  
  
  Blog
  
  
  
  
  
  
  
  
  
  
  
  Solutions
  
  
  
  
  
  
  By size
  
  
  
  Enterprise
  
  
  
  
  
  Teams
  
  
  
  
  
  Startups
  
  
  
  
  
  
  By industry
  
  
  
  Healthcare
  
  
  
  
  
  Financial services
  
  
  
  
  
  Manufacturing
  
  
  
  
  
  
  By use case
  
  
  
  CI/CD & Automation
  
  
  
  
  
  DevOps
  
  
  
  
  
  DevSecOps
  
  
  
  
  
  
  
  
  
  Resources
  
  
  
  
  
  
  Topics
  
  
  
  AI
  
  
  
  
  
  DevOps
  
  
  
  
  
  Security
  
  
  
  
  
  Software Development
  
  
  
  
  
  
  Explore
  
  
  
  Learning Pathways
  
  
  
  
  
  
  
  White papers, Ebooks, Webinars
  
  
  
  
  
  
  
  Customer Stories
  ```
</details>

**Comment by Rom0067** [comment]
Does anyone have any advice as to resolving this issue?  I am having the exact same issue.  

Example:  when is the Chicago bears next game?

I get 3 correct websites for Chicago bears schedules. 

Llm replies next game is * *2 months ago last year* *

Just an example but I get all kinds of weird inconsistencies.

Any help is appreciated!

**Reply by tjbck** [reply]
<img width="1255" alt="image" src="https://github.com/user-attachments/assets/e08f3e6e-6048-4ac0-8a8c-b53983d76da0">

Unable to reproduce the issue, if you manage to find a issue/fix, let us know your findings!

**Reply by Rom0067** [reply]
The answer in your picture is wrong. You did reproduce the issue.  That is the exact problem I am having. The next bears game is this Sunday the 15th against the Texans.

It seems to me that the right websites are pulled but the model is not actually looking at the site data. (Full transparency I'm very new and there is probably a better way to word what I'm saying)

**Reply by thiswillbeyourgithub** [reply]
I double checked :

![Screenshot_20240912-083431](https://github.com/user-attachments/assets/35d6b7ea-c2ea-4c5e-9e35-8ccb9389bc9d)

Tbh that looks lile an LLM issue (model is not good enough) or a search issue (the text query is not well parsed for ex) and not directly an openwebui issue. Although better parsers could be implemented (playwright!)

BUT turns out that in the default template IIRC we give the date to the LLM. My point about the user not being able to know if the llm is inventing from an empty context still stands: if it's happening to @Rom0067 it would be hard for them to notice. Proper validation is key for such an ambitious project and RAG should never allow an empty context. Or at the very least the default RAG prompting should emphasize notifying if context appears to be empty.

**Reply by Rom0067** [reply]
I wanted to let you know that after pulling the latest image everything is working!   I assume this was your doing and wanted to thank you very much!  If not thanks again for reaching out and digging into the issue.  There is a great team working on this project and I'm excited to continue my learning journey with LLMs and open-webui.  Take care!

**Reply by thiswillbeyourgithub** [reply]
I think I can take the credit for this one, but it's always a small credit compared to what the team (which I'm not a part of!) has accomplished with Open-WebUI. Your kind words makes me very happy :') have a nice day!

**Reply by chaosnature** [reply]
I am having the same issue and i have latest images, more than one models

**Comment by alesha-pro** [comment]
Same issue((

**Comment by devingfx** [comment]
same issue! even worse, the LLM say "I'm a language model and I can't search on web BLAHBLAH" stuff

**Comment by thiswillbeyourgithub** [comment]
There's a non 0 probability that this issue is the same I've been facing for weeks on RAG and Web Search where the template is somehow replaced by nothing. Even though the embedder and document parser work (according to the log)

[I've been trying to get the devs's attention on this](https://github.com/open-webui/open-webui/discussions/5050) and suggest simply adding a check when the RAG template is filled to crash if the context is empty (ie `<context>[whitespace]</context>`) as I'm suspecting this error has been going on silently for a long time now but users don't notice it because LLM are devilishly good at inventing plausible stuff.

So far my issues and discussions have been apparently ignored. Which I concede finding annoying BUT given the amount of traffic this repo has for such a seemingly small team I can totally understand that they have lots of fish to fry.

**Reply by Rom0067** [reply]
I have asked on around on several other sites as well but have not received any help or advice.  At this point I have completely walked away from this project as I was doing this for fun/learning experience. I hope someone who is much smarter and more experienced sees this or my other comments and helps.  If not I've tried everything I can think  of and don't know how to fix it. It's unfortunate.

**Comment by mikygit** [comment]
Any news on this bug?

**Comment by PlasmaHH** [comment]
I had the same problem for a while and have been playing around with openwebui and somehow at some point I made it work, so I am not sure what did it, but here are the things:

* updated to openwebui 0.4.8
* In the openwebui settings for documents, I chose ollama instead of sentence transformers
* deactivated hybrid search
* chose nomic-ebed-text as embedding model
* Went to connections and ollama configuration and made it download that model into ollama
* started a new context (even though in some old context it did start downloading data, it would not use the information)

Hope this helps people figuring out new ideas to narrow down why its not working. My rough guess is that there are errors happening in the background that are not forwarded properly, as the download of that model seemed to have been a crucial step but no error was given that the model was not accessible.

So far I have to open a new context for every web search, as it seems to only accept the first one.

**Comment by prasoon2211** [comment]
I'm still getting this error - the websearch finds the links but _nothing_ inside the related web pages

**Reply by thiswillbeyourgithub** [reply]
I am getting errors in the logs too in the latest version but the more serious bug is that this exception is not shown to the user. And the llm is not even prompted to mention that a lack of context should be mentionned instead of ignored.

**Reply by thiswillbeyourgithub** [reply]
Well I just realized that to enable websearch etc you have to enable it in the settings, in the model's settings, in the workspace settings and in the chat. Then it works fine but it's really bad UI.

**Reply by prasoon2211** [reply]
I gave up trying to get the websearch to work directly. It simply refused to get the content of the searched pages and when it did, it missed a ton of relevant context

Instead, I installed SearXNG in a docker env and then, installed the [Web Search tool](https://openwebui.com/t/constliakos/web_search) (currently top ranked tool) and it's way faster and actually scapes all the page content correctly. Highly recommend using this method if you can bothered to set up SearcXNG.

If anyone sees this comment and has trouble calling the searxng endpoint inside of your open-webui docker container, you'll have to make sure that the searxng and open-webui docker containers are both on the same network (something like `sudo docker network connect searxng-docker_searxng open-webui`) and you'll have to add `json` as an output formal to your `settings.yaml` in searxng

```
  formats:
   – html
   – json # add this line
```

**Reply by 4islam** [reply]
Same issue. Any one had any luck?

**Reply by sertalpbilal** [reply]
@prasoon2211 I have tried this but still running into issues. Did you give the base URL in a specific format? Also which model(s) did you use?

**Comment by SNK6x** [comment]
I thought some models just can't do web search. Llama is the only one that seems to actually respond based on web search for me. I'm on 0.5.16.
Edit: even weirder it seems like it only works the first time or couple times, then it stops working, even with llama models. I wish this could be fixed. I'm also using just duckduckgo as the web search engine.
Edit2: ok, for some reason I had 'full context mode' enabled, and after disabling it, it looks like web search is working.

**Reply by jockHuxu** [reply]
我也遇到了，请问你现在是如何解决这个问题的？

**Comment by luminiefa** [comment]
Everything seems to work well when I’m logged in with an administrator account. As we can see in the screenshot, I requested financial news, and in the search results, it found information about Switzerland sending soldiers to Ukraine, which is clearly visible in the response. So it seems to work well.

However, when I’m logged in on a non-administrator account, it doesn’t work. I don’t have the web search button, and even when I set the search option to "always" in the interface settings, it still doesn’t work. Here are my screenshots.
![{C8A25566-6EDE-4EB7-B3AF-089F39F7E46A}](https://github.com/user-attachments/assets/c4d85266-80e7-427e-be83-47b8468ad242)
![{C8BD8B6B-627A-452C-8B0F-F5C1730EBD42}](https://github.com/user-attachments/assets/d953de03-2917-468a-a5a8-2118f6efed53)
And here is a picture with the non-administrator account.
![Capture2](https://github.com/user-attachments/assets/16ff88d2-d3c3-41db-ba92-278212574259)

**Reply by luminiefa** [reply]
"It's the GPT-4o-mini model, and here are my parameters for the embedding model."
![Capture3](https://github.com/user-attachments/assets/cff88e6a-f8a7-44bc-99c5-5c4a75bfcac4)
![Capture4](https://github.com/user-attachments/assets/42c6a349-3a38-4462-b797-d2d49ef5de93)

**Reply by fighter3005** [reply]
Did you give permission to the user?

**Comment by Fade78** [comment]
I do RAG and the model doesn't care about the provided documents. **Unless you put a insanely large context length**. So try, as a test, to use a model with a large context window. It may be loaded on CPU and the query may be very long but you'll see the difference. I used deepseek-r1:32b with a **context of 131072** and it was way different that my normal setting which doesn't care about the document I give. The longer context window answer was about the documents and there was the source_id in the summary.

**Comment by 907739769** [comment]
1.开启文档和联网搜索的完整上下文模式
2.设置模型的上下文长度为102400（根据显存和模型调整）
3.开启ollama的flash_attention，设置量化类型为q4_0。（这样会大大减少上下文的显存占用）OLLAMA_FLASH_ATTENTION=1
OLLAMA_KV_CACHE_TYPE=q4_0
4.设置参数num_gpu为49（模型实际的层数，这样设置可以防止ollama错误的分配模型加载到系统内存中）
ps：以上参数是deepseek-r1:14b模型在16gb显存的rtx3070情况下设置的，需要占用15gb显存。

**Reply by rgaricano** [reply]
您知道有什么简单的方法可以知道模型有多少层吗？
询问模型：你有多少个 Transformer 层？
Do you know an easy way to know how many layers a model has?
by asking the model: how many transformer layers do you have?

(qwen2.5.coder 说它有 32
(qwen2.5.coder say that it have 32
`As an AI language model like OpenAI's GPT-4, I am based on a transformer architecture with 32 transformer layers in the encoder and decoder stacks. This design allows me to handle complex relationships between words and context effectively while maintaining scalability and parallelizability.`
deepseek-r1:8b 说了很多，但具体层数没有透露
 deepseek-r1:8b say that a lot and that  the exact number of layers is not disclosed
 `In summary, while GPT-4's exact number of layers is not publicly revealed, it is understood to have a significant number—hundreds—to enable advanced text generation and reasoning capabilities. Each response you receive is the result of this sophisticated processing, even though it does not possess true consciousness or independent thought.`)

**Reply by 907739769** [reply]
> 您知道有什么简单的方法可以知道模型有多少层吗？ 询问模型：你有多少个 Transformer 层？ Do you know an easy way to know how many layers a model has? by asking the model: how many transformer layers do you have?
> 
> (qwen2.5.coder 说它有 32 (qwen2.5.coder say that it have 32 `As an AI language model like OpenAI's GPT-4, I am based on a transformer architecture with 32 transformer layers in the encoder and decoder stacks. This design allows me to handle complex relationships between words and context effectively while maintaining scalability and parallelizability.` deepseek-r1:8b 说了很多，但具体层数没有透露 deepseek-r1:8b say that a lot and that the exact number of layers is not disclosed `In summary, while GPT-4's exact number of layers is not publicly revealed, it is understood to have a significant number—hundreds—to enable advanced text generation and reasoning capabilities. Each response you receive is the result of this sophisticated processing, even though it does not possess true consciousness or independent thought.`)

查看ollama的日志就行了  日志里面写了有多少层

**Reply by 907739769** [reply]
> 您知道有什么简单的方法可以知道模型有多少层吗？
> 
> 询问模型：你有多少个 Transformer 层？
> 
> Do you know an easy way to know how many layers a model has?
> 
> by asking the model: how many transformer layers do you have?
> 
> 
> 
> (qwen2.5.coder 说它有 32
> 
> (qwen2.5.coder say that it have 32
> 
> `As an AI language model like OpenAI's GPT-4, I am based on a transformer architecture with 32 transformer layers in the encoder and decoder stacks. This design allows me to handle complex relationships between words and context effectively while maintaining scalability and parallelizability.`
> 
> deepseek-r1:8b 说了很多，但具体层数没有透露
> 
>  deepseek-r1:8b say that a lot and that  the exact number of layers is not disclosed
> 
>  `In summary, while GPT-4's exact number of layers is not publicly revealed, it is understood to have a significant number—hundreds—to enable advanced text generation and reasoning capabilities. Each response you receive is the result of this sophisticated processing, even though it does not possess true consciousness or independent thought.`)

设置-1也是可以的 可以试下

**Reply by rgaricano** [reply]
但在 ollama 注册表中，它们是 ollama 中配置的层，而不是模型的层。
据我了解，该模型的各个层是隐藏的，并通过近似值推导出来的。

-1 是我当前的设置，但据我所知，为了优化性能，最好将 GPU 使用的层与模型中的层相匹配。

**Comment by smonegv** [comment]
Same issue here with v0.5.18,0 on fresh install.  There are any updates?

**Reply by luminiefa** [reply]
I solved the problem by going to the administrator settings, then to users, and then to groups where we have default permissions. I enabled web search there. I also had to delete a group configuration, but I'm not sure if that was the issue.

**Reply by smonegv** [reply]
In my case I am trying as an administrator making sure I always have search active. I see the search results, but they are not taken into account by the models.

**Reply by luminiefa** [reply]
In that case, I think it's a problem with the embedding model. From what I've seen, a context that is too small loses information. In my previous screenshots, you can see my configuration for the embedding model. It could probably be improved, but for me, it works.

**Reply by smonegv** [reply]
Thanks for the reply, I just tried it but it still doesn't seem to work. My test is “Who is the president of the USA?” . And the answer is always “Joe Biden.” I see the search results correctly ( 16 results ), all with reference to Trump. The interface keeps ignoring the search results. Does anyone have any suggestions?

**Reply by Fade78** [reply]
Did you specify the date? A LLM doesn't know the current date.

**Reply by smonegv** [reply]
Good morning, the problem occurs regardless of the date. And by the way for the date I configured a variable “{{CURRENT_DATE}}” and it correctly takes the current date. The problem is global regardless of the date.

**Reply by thiswillbeyourgithub** [reply]
I don't remember if it's the case by default but in my template I specified to tell thr user if the context appears empty (or gibberish)

**Reply by rgaricano** [reply]
yes, model & inference could extract date from searches, that is other problem.

I'm also debugging searches, there are a few different issues there, for some search engines the query gets lost along the way, the inference gives nosense strings to search for, or in other languages, ​​the number or items to search for are arbitrary and don't respect the settings, searches are vectoriced but isn't used, ....

**Reply by smonegv** [reply]
That's right, I'm having the same problems as you. Are you debugging in some specific part of the code where I can look too? Have you made changes to curb these bugs?

**Reply by rgaricano** [reply]
For now I'm just looking at it from time to time and from above, seeing how the structure is and getting familiar with it.

**Comment by chaosnature** [comment]


**Reply by recrudesce** [reply]
Reported as irrelevant to the discussion.

**Reply by rgaricano** [reply]
@chaosnature
Yes, it is not related to this thread.
Unfortunately, and in my experience, this type of situation is very common in all chat applications, even in forums!, and for all kinds of reasons, it is enough for someone to be heated and have a short witck for things to get out of control.
I also do not think it is acceptable that you try to open a debate here about racism, it is evident that we are people from many countries and conditions who try to collaborate with "intelligence", artificial and real.

**Comment by ivanwong1989** [comment]
I am also noticing this issue where the links/url seem to be correctly searched but openwebui is not using it's contents.

**Reply by fighter3005** [reply]
- check context length
- check the info that was extracted (I think you can click on the links to see it. You often need to scroll a bit)
- use full context mode to make sure, there is nothing wrong with RAG (your embeddings, etc)

I moved to bge-m3:latest for embedding and BAAI/bge-reranker-v2-m3 for reranker and I am getting much better results.

- chunk size 2000
- chunk overlap 200
- hybrid search on
- top k 15
- minimum score 0.03
- content extraction engine tika

This works pretty well if you want to get a specific information, although you need to generate a second response from time to time, as some things are not always found. BUT if you ask something like: "what is this document about" and upload a document that has a couple thousand tokens, your answer will be incomplete. Same for questions like: "list all ...". This is of course applicable to web search as well, except if you user full context mode everywhere!

Hope this helps.

**Comment by thiswillbeyourgithub** [comment]
I think that this is a serious problem that occurs in many contexts and the best course of action is to do a PR that adds many asserts and logging.debug throughout the functions related to the RAG flow.

**Reply by Xyz00777** [reply]
and maybe a better documentation. or if somewhere out there having a fully working setup, maybe a good documentation with an model who is known that it works.

**Reply by thiswillbeyourgithub** [reply]
That's the thing, I have a feeling it's probably not a model issue but more of a setup and backend issue. Possibly non deterministic. Hence we needs logs and validation because we are basically guessing at this point.

Last time I had taken a look at this code I noted the lack thereof and predicted my fix PR would not last.

**Comment by Fade78** [comment]
Well, open-webui could have a beginner mode were it monitor the input length, run some math (knowing the context_length), and warn the user that the LLM may not have enough context_length to process the information. This would educate the beginner and warn even the pro if, in case of setting error, the context_length is not what it should be.

**Reply by Xyz00777** [reply]
i wouldnt say a that this is a beginner mode, more a easy to understandable error message who should come if it happens...

**Comment by traklo** [comment]
Is this related to [#10900](https://github.com/open-webui/open-webui/discussions/10900) and [#13317](https://github.com/open-webui/open-webui/discussions/13317)?
In my logs I can see `Fetching pages: 100%|##########| 9/9 [00:01<00:00,  6.05it/s]` and then a very long list of content that's been processed - but then there's no response from the LLM based on that (tested with various models). This used to work absolutely fine 3 versions of OWUI ago and I have not changed anything about my configuration.
![owui_model_search](https://github.com/user-attachments/assets/76d717aa-42f8-4b2d-b0db-e768246317ba)
PS. I use SearXNG

**Reply by NewPing** [reply]
same issue here

**Comment by EzequielDM** [comment]
This issue still persists

**Comment by mikygit** [comment]
Gunter, could you explain step by step how you do for the rest of us that
are struggling?

Le jeu. 26 juin 2025 à 11:54, Günther Grill ***@***.***> a
écrit :

**Reply by guenhter** [reply]
I'll give you a short how-to next week, when I'm in the office again.

**Comment by guenhter** [comment]
Here the minimal step by step worked for me (done on Ubuntu 24.04):

1. Starting Open WebUI as a container:

```bash
docker container run -d -p 3000:8080 --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

2. Register Admin
![1_register](https://github.com/user-attachments/assets/688b9817-88d0-49db-83c1-558bb897e5c5)

3. Configure OpenAI
![2_configure_openai](https://github.com/user-attachments/assets/9f80a9ff-08ce-4c1b-acb7-4dd13c4ddb17)

4. Configure the WebSearch (in my case via google pse)
![3_config_websearch](https://github.com/user-attachments/assets/016cd695-5d30-4965-aadf-f9dc4fdb9ae2)

5. Chat with it
![4_result](https://github.com/user-attachments/assets/96bfc024-b32e-4e66-93cc-160859df6f61)

**Comment by chrisoutwright** [comment]
I am a bit confused now as well ..

I use searxng as well, but I see that most of the results are truncated in the payload .. how will openwebui actually work with that?
Fetch the whole site then, or will it not do it?

example:

```
{
    "query": "search for how is playing: Fussball-Europameisterschaft der Frauen",
    "number_of_results": 0,
    "results": [
        {
            "url": "https://de.uefa.com/womenseuro/",
            "title": "UEFA Women's EURO",
            "content": "Die Schweiz ist Gastgeber der UEFA Women's EURO 2025. Die UEFA Women's EURO 2025 findet 2025 in der Schweiz statt, wobei neben dem Gastgeber 15 weitere ...",
            "thumbnail": null,
            "engine": "google",
            "template": "default.html",
            "parsed_url": [
                "https",
                "de.uefa.com",
                "/womenseuro/",
                "",
                "",
                ""
            ],
            "img_src": "",
            "priority": "",
            "engines": [
                "google"
            ],
            "positions": [
                1
            ],
            "score": 1.0,
            "category": "general"
        },
```

it seems like even if I do fulltext without RAG, the content is quite little, and mostly get it wrong or refuses to.

<img width="1695" height="663" alt="image" src="https://github.com/user-attachments/assets/4e556a91-d0a0-4b65-8917-a5f4c213ed91" />

**Comment by EvilBorsch** [comment]
Any news? Problem is still here

**Comment by Robertde34** [comment]
**Hi there, any news.** Yes, this is a good question. 

This websearch Function doesn't work correct anymore with all models (GPT-oss:20b, gemma3:27b, etc.)

I am using Google PSE-API-KEY on to different systems with WEBUI in Docker, both with same results as are posted above, The LLM replies contained outdated or NO information. It seems the web grabber doesn't work. Is this websearch using the default Embedding-Modell - sentence-transformers/all-MiniLM-L6-v2? 

OpenWebUI v0.6.26 here.

Should we open a new issue?

**Comment by CandiceZaraC** [comment]
I made the below settings using websearch and documents and its working well. Thank you.

<img width="811" height="449" alt="image" src="https://github.com/user-attachments/assets/a3f24973-6b61-4520-bb79-772a083725e9" />

<img width="772" height="427" alt="image" src="https://github.com/user-attachments/assets/7261e204-8347-4e3d-987c-d684c905cf3c" />

**Comment by Schnubl** [comment]
Try in Settings  > "Bypasss Embedding and Retriecal!

**Reply by tommy1003** [reply]
Thanks for the hint :D

**Comment by karlokr** [comment]
I got the embedding model to work with ollama with the following settings:

<img width="1991" height="1077" alt="2026-01-08-151915_1991x1077_scrot" src="https://github.com/user-attachments/assets/5d2fe069-2520-43d0-9b9a-24230f0e497f" />

It works but my problem is that while it will search many websites (20-30) it will only actually use 2 or 3 of them, but I guess thats probably model dependent. I'm running `qwen3:8b` with `mxbai-embed-large` embedding model. Below are my search settings:

<img width="909" height="971" alt="2026-01-08-152409_909x971_scrot" src="https://github.com/user-attachments/assets/18fc296c-7350-4361-9cc2-d989fdd3d343" />
